python/ray/_private/runtime_env/conda.py

```diff
@@ -216,27 +270,22 @@ def setup(self,
                 extra_pip_dependencies = []
             conda_dict = inject_dependencies(conda_dict, _current_py_version(),
                                              extra_pip_dependencies)
+
             logger.info(f"Setting up conda environment with {runtime_env}")
-            # It is not safe for multiple processes to install conda envs
-            # concurrently, even if the envs are different, so use a global
-            # lock for all conda installs.
-            # See https://github.com/ray-project/ray/issues/17086
-            file_lock_name = "ray-conda-install.lock"
-            with FileLock(os.path.join(self._resources_dir, file_lock_name)):
```

Comment:
do we not still need the file lock logic?

Reply:
I think this is leftover from the days when each worker installed the runtime env at the same time.  Now that all env creation is going through the agent, and the agent isn't async, we don't need the lock anymore, right?

Reply:
The agent is still async and installing things in background threads so I think we need it for now

Reply:
we should do this coordination in the agent/plugins themselves instead of using filelocks in the near future though

Reply:
Oh you're totally right... Hopefully this explains why a few of the existing tests are failing